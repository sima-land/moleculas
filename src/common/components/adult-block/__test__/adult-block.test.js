import React from 'react';
import { mount } from 'enzyme';
import AdultBlock from '../adult-block';
import Link from '@dev-dep/ui-nucleons/link';

describe('<AdultBlock />', () => {
  it('should render without props', () => {
    const wrapper = mount(<AdultBlock />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render with props', () => {
    const onClick = jest.fn();
    const wrapper = mount(
      <AdultBlock
        isAuthUser
        onClick={onClick}
      />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.find(Link).at(0).prop('onClick')();
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
