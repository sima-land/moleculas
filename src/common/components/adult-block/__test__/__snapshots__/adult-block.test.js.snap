// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdultBlock /> should render with props 1`] = `
<AdultBlock
  isAuthUser={true}
  onClick={[MockFunction]}
>
  <Box
    direction="column"
    display="flex"
    flex="grow"
    marginTop={0}
  >
    <div
      className="display-flex flex-grow direction-column M-t__0"
    >
      <Box
        marginBottom={1}
      >
        <div
          className="M-b__1"
        >
          <ForwardRef(Text)
            size={16}
            weight={600}
          >
            <span
              className="size-16 line-height-24 weight-600"
            >
              Товар для взрослых
            </span>
          </ForwardRef(Text)>
        </div>
      </Box>
      <ForwardRef(Text)
        lineHeight={20}
        size={12}
      >
        <span
          className="size-12 line-height-20"
        >
          Подтвердите, что вы старше 18 лет, чтобы посмотреть товар.
        </span>
      </ForwardRef(Text)>
      <Box
        marginTop={6}
      >
        <div
          className="M-t__6"
        >
          <ForwardRef(Link)
            onClick={[MockFunction]}
            pseudo={true}
          >
            <a
              className="link color__brand-blue hover-color__brand-deep-blue"
              href="javascript:;"
              onClick={[MockFunction]}
              role="button"
              tabIndex={0}
            >
              <ForwardRef(Text)
                lineHeight={32}
                size={16}
                weight={600}
              >
                <span
                  className="size-16 line-height-32 weight-600"
                >
                  Подтвердить
                </span>
              </ForwardRef(Text)>
            </a>
          </ForwardRef(Link)>
        </div>
      </Box>
    </div>
  </Box>
</AdultBlock>
`;

exports[`<AdultBlock /> should render without props 1`] = `
<AdultBlock>
  <Box
    direction="column"
    display="flex"
    flex="grow"
    marginTop={0}
  >
    <div
      className="display-flex flex-grow direction-column M-t__0"
    >
      <Box
        marginBottom={1}
      >
        <div
          className="M-b__1"
        >
          <ForwardRef(Text)
            size={16}
            weight={600}
          >
            <span
              className="size-16 line-height-24 weight-600"
            >
              Товар для взрослых
            </span>
          </ForwardRef(Text)>
        </div>
      </Box>
      <ForwardRef(Text)
        lineHeight={20}
        size={12}
      >
        <span
          className="size-12 line-height-20"
        >
          Если вам уже есть 18 лет, авторизуйтесь, чтобы перейти к товару.
        </span>
      </ForwardRef(Text)>
      <Box
        marginTop={6}
      >
        <div
          className="M-t__6"
        >
          <ForwardRef(Link)
            pseudo={true}
          >
            <a
              className="link color__brand-blue hover-color__brand-deep-blue"
              href="javascript:;"
              role="button"
              tabIndex={0}
            >
              <ForwardRef(Text)
                lineHeight={32}
                size={16}
                weight={600}
              >
                <span
                  className="size-16 line-height-32 weight-600"
                >
                  Авторизоваться
                </span>
              </ForwardRef(Text)>
            </a>
          </ForwardRef(Link)>
        </div>
      </Box>
    </div>
  </Box>
</AdultBlock>
`;
